torch/csrc/api/src/nn/modules/embedding.cpp

```diff
@@ -11,27 +11,203 @@
+      TORCH_CHECK(offsets.dim() == 0, "offsets[0] has to be 0, i.e., the first sequence in the mini-batch has to start from position 0.
+                However, got ", int(offsets[0]), "\n");
+      TORCH_CHECK(int(offsets[offsets.size(0)-1]) <= input.size(0), "offsets[offsets.size(0)-1] can not be greater than input's length({)",
+                input.size(0), "}), but got offsets[offsets.size(0)-1] of {", int(offsets[offsets.size(0)-1]), "}\n");
+      }
+    }
+    else{
+      TORCH_CHECK(false, "input has to be 1D or 2D Tensor,but got Tensor of dimension {", input.dims(), "}\n");
+    }
+
+    int mode_enum;
+    if(mode == "sum") {
+      mode_enum = 0;
+    }
+    else if(mode == "mean") {
+      mode_enum = 1;
+    }
+    else if(mode =="max") {
+      mode_enum = 2;
+      TORCH_CHECK(!options.scale_grad_by_freq(), "max mode does not support scaling the gradient by the frequency");
+      TORCH_CHECK(!sparse, "max mode does not support sparse weights");
+    }
+    else{
+      TORCH_CHECK(false, "mode has to be one of sum, mean or max");
+    }
+
+    if(options.max_norm() != c10::nullopt) {
+      torch::NoGradGuard no_grad;
+      torch::embedding_renorm_(weight, input, *options.max_norm(), options.norm_type());
+    }
+
+    TORCH_CHECK((per_sample_weights == c10::nullopt) || (mode == "sum"), "embedding_bag: per_sample_weights was not null
+          per_sample_weights is only supported for mode='sum' (got mode='{", mode,
+          "})Please open a feature request on GitHub.");
+    return torch::embedding_bag(weight, input, offsets, scale_grad_by_freq, mode_enum, sparse, per_sample_weights);
+  }
+
+  void EmbeddingBag::pretty_print(std::ostream& stream) const {
+    stream << "torch::nn::EmbeddingBag(num_embeddings=" << options.num_embeddings()
+           << ", embedding_dim=" << options.embedding_dim();
+    if(options.max_norm() != c10::nullopt) {
+      stream << ",max_norm=" << *options.max_norm();
+    }
+    if(options.norm_type() != 2) {
+      stream << ",norm_type=" << options.norm_type();
+    }
+    if(options.scale_grad_by_freq()) {
+      stream << ",scale_grad_by_freq=" << options.scale_grad_by_freq();
+    }
+    stream << ",mode="<<mode<<")";
+  }
+
+  EmbeddingBagImpl& EmbeddingBagImpl::from_pretrained(Tensor embeddings, bool freeze = true, c10::optional<float> mex_norm = c10::nullopt,
+     float norm_type = 2., bool scale_grad_by_freq = false, string mode = "sum", bool sparse = false) {
+      TORCH_CHECK(embeddings.dim() == 2, "Embeddings parameter is expected to be 2-embedding_dimal");
+      EmbeddingImpl embeddingbag = EmbeddingImpl(
+              num_embeddings=embeddings.size(0),
+              embedding_dim=embeddings.size(1),
+              _weight=embeddings,
+              max_norm=max_norm,
+              norm_type=norm_type,
+              scale_grad_by_freq=scale_grad_by_freq,
+              mode=mode,
+              sparse=sparse);
```

Comment:
We need to pass the EmbeddingBag options to the constructor, because keyword argument is not supported in C++.